compiler/rustc_middle/src/query/keys.rs

```diff
@@ -45,14 +45,17 @@ pub trait Key: Sized {
     }
 }
 
+pub trait AsLocalKey: Key {
```

Comment:
So that the `as_local_key` only has to be provided for the keys where we actually make use of it. This should make it a little easier to make sure that all of the impls are correct and not run into things like #109109 again.

Reply:
Yeah, exactly what Nils said.